indispensable-josef/src/commonMain/kotlin/at/asitplus/signum/indispensable/josef/JweEncryption.kt

```diff
@@ -15,43 +18,51 @@ import kotlinx.serialization.encoding.Encoder
  * and also [RFC 7518](https://datatracker.ietf.org/doc/html/rfc7518)
  */
 @Serializable(with = JweEncryptionSerializer::class)
-enum class JweEncryption(val text: String) {
-
-    A128GCM("A128GCM"),
-    A192GCM("A192GCM"),
-    A256GCM("A256GCM"),
-    A128CBC_HS256("A128CBC-HS256"),
-    A192CBC_HS384("A192CBC-HS384"),
-    A256CBC_HS512("A256CBC-HS512")
+enum class JweEncryption(val identifier: String, val algorithm: SymmetricEncryptionAlgorithm<*, *, *>) {
```

Comment:
this should probably have its own `Specialized...` interface

Reply:
probably, yes